repository/avail-light.repository.ts

```diff
@@ -48,8 +49,9 @@ export async function runLC(onBlock: Function, registerUnsubscribe: Function): P
         //Create required info for process block
         const block: Block = { number: blockNumber, hash: blockHash, totalCellCount: totalCellCount, confidence: 0, sampleCount: sampleCount, timestamp }
         const matrix: Matrix = { maxRow: r, maxCol: c, verifiedCells: [], totalCellCount }
-        onBlock(block, matrix, randomCells, proofs, commitments)
+        onBlock((list: BlockToProcess[]) => [...list, { block, matrix, randomCells, proofs, commitments }])
```

Comment:
We need to separate the concerns better. This should either be a method, that simply notifies the app on new blocks to process it, or process it within this subscription, and notify the app on a processed block becoming available. If two different methods are accessing the same array, it leads to un explained behaviours as the code becomes more complicated moving forward. So let this only return the block and other details required for processing it, and let the app handle the array.

Reply:
Yes, I completely understand your point, will change this